pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -15,9 +15,104 @@
 This submodule defines functions to decompose controlled operations
 """
 
+import typing
+import numpy as np
+import numpy.linalg as npl
 import pennylane as qml
 from pennylane.operation import Operator
 from pennylane.wires import Wires
+from pennylane import math
```

Comment:
Since we no longer have the decomposition as differentiable, we can just use vanilla numpy everywhere instead. That will reduce some overheads that come with using `pennylane.math`.

Reply:
Do we really need to revert the change from `np` to `math` when we might need it later if we decide it needs to be differentiable? It works as is, I'd say keep it unless there's a reason to change it. I also don't think it's a meaningful optimization, it's a very small constant cost, I don't think anyone would notice the difference.

Reply:
> Since we no longer have the decomposition as differentiable

Where do we lose differentiability?

Reply:
In the general algorithm, where it does eigendecomposition. Also some other places like the B matrix computation, but the eigendecomposition is the one that will be most difficult to make differentiable.

There is this paper, perhaps we will add it in the future: https://arxiv.org/pdf/1906.09023v2.pdf

Reply:
uses np now. added safety in compute B matrix for negative sqrt case